d2ir/pattern_test.go

```diff
@@ -314,21 +339,355 @@ task.** -> fast
+				assertQuery(t, m, 24, 0, nil, "")
+				assertQuery(t, m, 0, 0, "page", "a.shape")
+				assertQuery(t, m, 0, 0, "page", "b.shape")
+				assertQuery(t, m, 0, 0, "page", "c.shape")
+				assertQuery(t, m, 0, 0, "page", "d.shape")
+				assertQuery(t, m, 0, 0, "page", "scenarios.x.p.shape")
+				assertQuery(t, m, 0, 0, "page", "layers.x.p.shape")
+			},
+		},
+		{
+			name: "triple-glob/edge-defaults",
+			run: func(t testing.TB) {
+				m, err := compile(t, `(*** -> ***)[*]: {
+	target-arrowhead.shape: diamond
+}
+
+a -> b
+c -> d
+
+scenarios.x: { p -> q }
+layers.x: { j -> f }
+`)
+				assert.Success(t, err)
+				assertQuery(t, m, 28, 6, nil, "")
+				assertQuery(t, m, 0, 0, "diamond", "(a -> b)[0].target-arrowhead.shape")
+				assertQuery(t, m, 0, 0, "diamond", "(c -> d)[0].target-arrowhead.shape")
+				assertQuery(t, m, 0, 0, "diamond", "scenarios.x.(a -> b)[0].target-arrowhead.shape")
+				assertQuery(t, m, 0, 0, "diamond", "scenarios.x.(c -> d)[0].target-arrowhead.shape")
+				assertQuery(t, m, 0, 0, "diamond", "scenarios.x.(p -> q)[0].target-arrowhead.shape")
+				assertQuery(t, m, 4, 1, nil, "layers.x")
+				assertQuery(t, m, 0, 0, "diamond", "layers.x.(j -> f)[0].target-arrowhead.shape")
+			},
+		},
+		{
+			name: "alixander-review/1",
+			run: func(t testing.TB) {
+				m, err := compile(t, `
+***.style.fill: yellow
+**.shape: circle
+*.style.multiple: true
+
+x: {
+  y
+}
+
+layers: {
+  next: {
+    a
+  }
+}
+`)
+				assert.Success(t, err)
+				assertQuery(t, m, 14, 0, nil, "")
+			},
+		},
+		{
+			name: "alixander-review/2",
+			run: func(t testing.TB) {
+				m, err := compile(t, `
+a
+
+* -> y
+
+b
+c
+`)
+				assert.Success(t, err)
+				assertQuery(t, m, 4, 3, nil, "")
+			},
+		},
+		{
+			name: "alixander-review/3",
+			run: func(t testing.TB) {
+				m, err := compile(t, `
+a
+
+***.b -> c
+
+layers: {
+  z: {
+    d
+  }
+}
+`)
+				assert.Success(t, err)
+				assertQuery(t, m, 8, 2, nil, "")
+			},
+		},
+		{
+			name: "alixander-review/4",
+			run: func(t testing.TB) {
+				m, err := compile(t, `
+**.child
+
+a
+b
+c
+`)
+				assert.Success(t, err)
+				assertQuery(t, m, 6, 0, nil, "")
+			},
+		},
+		{
+			name: "alixander-review/5",
+			run: func(t testing.TB) {
+				m, err := compile(t, `
+**.style.fill: red
+
+scenarios: {
+  b: {
+    a -> b
+  }
+}
+`)
+				assert.Success(t, err)
+				assertQuery(t, m, 8, 1, nil, "")
+				assertQuery(t, m, 0, 0, "red", "scenarios.b.a.style.fill")
+				assertQuery(t, m, 0, 0, "red", "scenarios.b.b.style.fill")
+			},
+		},
+		{
+			name: "alixander-review/6",
+			run: func(t testing.TB) {
+				m, err := compile(t, `
+(* -> *)[*].style.opacity: 0.1
+
+x -> y: hi
+x -> y
+`)
+				assert.Success(t, err)
+				assertQuery(t, m, 6, 2, nil, "")
+				assertQuery(t, m, 0, 0, 0.1, "(x -> y)[0].style.opacity")
+				assertQuery(t, m, 0, 0, 0.1, "(x -> y)[1].style.opacity")
+			},
+		},
+		{
+			name: "alixander-review/7",
+			run: func(t testing.TB) {
+				m, err := compile(t, `
+*: {
```

Comment:
this wasn't the test btw. it was "if i comment either of the globs out, it becomes blue". so it shoudl test that having each of these globs removed, the table fill is still red

Reply:
Yea I didn't fully understand your comment. There was a bug here but the bug was that it should always be blue because lazy globs do not override normal keys. Only globs after normal keys override them. So if you move one of the globs below table, then it'll work as you suggest.